File: Object-Oriented-Programming-January2015/06.EncapsulationAndPolymorphism/01.Shapes/Rectangle.cs
using System;

class Rectangle : BasicShape
{
    public Rectangle(double width, double height) : base(width, height)
    {
    }

    public override double CalculatePerimeter()
    {
        return this.Width * 2 + this.Height * 2;
    }

    public override double CalculateArea()
    {
        return this.Width * this.Height;
    }
}

File: Level 2/Object-Oriented-Programming-January2015/08.OOP-In-Java/src/_01_Geometry/PlaneShapes/Rectangle.java
package _01_Geometry.PlaneShapes;

import _01_Geometry.PlaneShape;
import _01_Geometry.Vertices.Vertex2D;

public class Rectangle extends PlaneShape {
	private double width;
	private double height;
	
	public Rectangle(Vertex2D vertex2d, double width, double height) {
		super(vertex2d);
		
		this.setWidth(width);
		this.setHeight(height);
	}
	
	public double getWidth() {
		return width;
	}

	public void setWidth(double width) {
		this.width = width;
	}

	public double getHeight() {
		return height;
	}

	public void setHeight(double height) {
		this.height = height;
	}

	@Override
	public double getPerimeter() {
		double perimeter = this.getWidth() * 2 + this.getHeight() * 2;
		return perimeter;
	}

	@Override
	public double getArea() {
		double area = this.getWidth() * this.getHeight();
		return area;
	}
}
